### ISSUE TITLE
feat(workflow): wire FILTER_ANTIFOLD into pipeline

### ISSUE THREAD

**Issue by crscarpinteiro** [JelPej/nf-core-antibody-pipeline#58]
## Summary

Once the `FILTER_ANTIFOLD` module (#36) is delivered, wire it into the main workflow between `ANTIFOLD_SPLIT` and `BIOPHI_SAPIENS`.

## Wiring

```nextflow
// Join ANTIFOLD_CDR logits with ANTIFOLD_SPLIT output on meta.id
ch_filter_antifold_input = ANTIFOLD_SPLIT.out.fasta
    .join(ANTIFOLD_CDR.out.logits, by: [0])

FILTER_ANTIFOLD(ch_filter_antifold_input)

BIOPHI_SAPIENS(FILTER_ANTIFOLD.out.filtered)
```

## Channel contract

| Boundary | Shape |
|---|---|
| ANTIFOLD_SPLIT → FILTER_ANTIFOLD | `tuple val(meta), path(split_fasta), path(scores_csv)` (join `ANTIFOLD_CDR.out.logits` on `meta.id`) |
| FILTER_ANTIFOLD → BIOPHI_SAPIENS | `tuple val(meta), path(filtered_fasta)` |

## Acceptance criteria

- [ ] `FILTER_ANTIFOLD` is called between `ANTIFOLD_SPLIT` and `BIOPHI_SAPIENS` in `workflows/antibodyoptimization.nf`
- [ ] Logits channel joined correctly on `meta.id`
- [ ] `conf/modules.config` entry added for `FILTER_ANTIFOLD` (publishDir, ext.args)
- [ ] Stub run passes all processes including `FILTER_ANTIFOLD`
- [ ] `copilot-instructions.md` issue tracker updated

## Blocked by

#36
